### PR TITLE
feat: add confirmation dialog for file removal

### PR DESCRIPTION
**Pull request by bobtista**
Added a modal popup that requires user confirmation before removing files in the File Manager. This prevents accidental file deletions by requiring explicit user confirmation.

### CODE REVIEW

**Comment by bobtista** [comment]
Screenshot 2024-12-09 at 12 30 48 PM

**Comment by bobtista** [comment]
Screenshot 2024-12-09 at 12 30 48 PM

File: src/imguiclient/imguiapp.cpp [hunk 1]
```diff
@@ -2391,10 +2391,36 @@ void ImGuiApp::FileManagerDescriptorActions(ProgramFileDescriptor &descriptor, b
...
+        }
+
+        // Center the modal popup
+        ImVec2 center = ImGui::GetMainViewport()->GetCenter();
+        ImGui::SetNextWindowPos(center, ImGuiCond_Appearing, ImVec2(0.5f, 0.5f));
+
+        if (ImGui::BeginPopupModal("Remove File?", NULL, ImGuiWindowFlags_AlwaysAutoResize))
+        {
+            ImGui::Text("Are you sure you want to remove this file?\nThis operation cannot be undone!\n\n");
+            ImGui::Separator();
```
Comment (xezon) [inline src/imguiclient/imguiapp.cpp]: Perhaps it looks better without this line.

File: src/imguiclient/imguiapp.cpp [hunk 2]
```diff
@@ -2391,10 +2391,36 @@ void ImGuiApp::FileManagerDescriptorActions(ProgramFileDescriptor &descriptor, b
...
+            ImGui::OpenPopup("Remove File?");
+        }
+
+        // Center the modal popup
+        ImVec2 center = ImGui::GetMainViewport()->GetCenter();
+        ImGui::SetNextWindowPos(center, ImGuiCond_Appearing, ImVec2(0.5f, 0.5f));
+
+        if (ImGui::BeginPopupModal("Remove File?", NULL, ImGuiWindowFlags_AlwaysAutoResize))
+        {
+            ImGui::Text("Are you sure you want to remove this file?\nThis operation cannot be undone!\n\n");
```
Comment (xezon) [inline src/imguiclient/imguiapp.cpp]: I suggest say "Are you sure you want to remove this file from the list?\nIt does not delete it from disk.\n\n"

File: src/imguiclient/imguiapp.cpp [hunk 3]
```diff
@@ -2391,10 +2391,36 @@ void ImGuiApp::FileManagerDescriptorActions(ProgramFileDescriptor &descriptor, b
...
 
-        // #TODO: Guard this button press with a confirmation dialog ?
-        // There is an example for this in "Dear ImGui Demo" > "Popups & Modal windows" > "Modals".
+        if (ImGui::Button("Remove"))
+        {
+            ImGui::OpenPopup("Remove File?");
+        }
+
+        // Center the modal popup
+        ImVec2 center = ImGui::GetMainViewport()->GetCenter();
```
Comment (xezon) [inline src/imguiclient/imguiapp.cpp]: I suggest move this popup code in a separate function.

File: src/imguiclient/imguiapp.cpp (hunk 2 above)
Comment (xezon) [inline src/imguiclient/imguiapp.cpp]: Use `ImGui::TextUnformatted`

**Comment by bobtista** [comment]
Screenshot 2024-12-12 at 8 25 14 AM

**Comment by xezon** [comment]
Could you perhaps use a text widget where it does auto line breaks? I have seen such functionality in ImGui (check the demo). Then make this text box fit the dimensions of the 2 buttons below.

**Comment by bobtista** [comment]
Screenshot 2024-12-12 at 12 46 35 PM

File: src/imguiclient/imguiapp.cpp [hunk 4]
```diff
@@ -2380,6 +2380,36 @@ void ImGuiApp::FileManagerDescriptorPdbConfig(ProgramFileDescriptor &descriptor)
     }
 }
 
+void ImGuiApp::ShowRemoveFileConfirmationPopup(bool &erased)
```
Comment (xezon) [inline src/imguiclient/imguiapp.cpp]: I think it would be better to use return bool.

File: src/imguiclient/imguiapp.cpp [hunk 5]
```diff
@@ -2380,6 +2380,36 @@ void ImGuiApp::FileManagerDescriptorPdbConfig(ProgramFileDescriptor &descriptor)
...
+{
+    // Center the modal popup
```
Comment (xezon) [inline src/imguiclient/imguiapp.cpp]: I think you can remove all comments in this function because code is self-explanatory.

**Review by xezon — COMMENTED** [review]
Please merge last state of parent branch.

**Review by xezon — COMMENTED** [review]
I tested this change and it does not work correctly. All files are immediately deleted after they have been added.

**Comment by bobtista** [comment]
I found and fixed a bug that was causing it to always remove the most recently added file instead of the one selected for removal. Then merged with latest and tested, it's working for me now. Can you try again?

**Review by xezon — APPROVED** [review]

**Pull request merged**